code/modules/projectiles/guns/projectile/constructable/swords.dm

```diff
@@ -24,6 +24,7 @@
 		return
 	to_chat(user, "You remove the hypospray from \the [src].")
 	var/obj/item/weapon/reagent_containers/hypospray/H = new (get_turf(user.loc))
+	H.reagents.remove_reagent("doctorsdelight", 30)
```

Comment:
I don't think it makes sense for it to retain the chemicals after being used to dispense reagents from the sword's beaker, but you make a good point about the fingerprints.